### PR TITLE
Inline link padding on default styling

### LINKED ISSUE #777: Inline link padding/margins do not scale when font size changes
**Describe the bug**
The link padding and margin should be in REMs not px - https://github.com/bbc/psammead/blob/latest/packages/components/psammead-inline-link/src/index.jsx#L24-L25

The impact of this currently is that padding and margins will not scale correctly as the user changes their font size in the browser.

**Expected behaviour**
The padding and margins should scale with font-size 

**Dev insight**
Note that snapshots will need updating.

### LINKED ISSUE #1024: Inline link styling new styling
**Is your feature request related to a problem? Please describe.**
When in hover/focus state the inline link with have a red background. With background-colour property only spanning the width of the element there is limited space to the left and right of the first and last character respectively. Because of this we should add padding and minus margin left an right to increase the legibility of the text in this state, especially when using italic text as it leans out of the elements bounding box. 

**Describe the solution you'd like**
Pair with UX to define the CSS logic as above, ensuring all a11y concerns are addressed. 

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Visual regression and snapshots

**Additional context**

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #1024
Resolves #777 

**Overall change:** Sets the background colour to red, text colour to white during hover and focus states. Adds left/right padding and minus margin to increase buffer between the glyphs and background colour for legibility reasons. Adds pre-wrap to enhance legibility on the end of a line when the link wraps to a new line. 

**Code changes:**

- Uses REMs for padding and minus margin to increase with root font-size
- Updates an incorrect PR number in changelog
- Publishes as alpha
- Updates snapshots
- Removes part of README as it is no longer relevant 

Testing notes: 

- Run the following command to:
`
npm run build && cp packages/components/psammead-inline-link/dist/index.js packages/components/psammead-caption/node_modules/@bbc/psammead-inline-link/dist/index.js && cp packages/components/psammead-inline-link/dist/index.js packages/components/psammead-paragraph/node_modules/@bbc/psammead-inline-link/dist/index.js && npm run storybook
`
- Test cross browser the paragraph and caption stories which have inline-link
- Also ensure [this bug](https://github.com/bbc/simorgh/pull/2258#pullrequestreview-258785470) cannot be reproduced

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-inline-link/CHANGELOG.md [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
 | 1.1.7 | [PR#892](https://github.com/bbc/psammead/pull/892) Bump dependencies |
 | 1.1.6 | [PR#942](https://github.com/bbc/psammead/pull/942) Revert new inline-link styling due to [bug](https://github.com/bbc/simorgh/issues/2370) |
 | 1.1.5 | [PR#783](https://github.com/bbc/psammead/pull/783) Update to latest psammead-test-helpers. Update snapshots. |
 | 1.1.4 | [PR#769](https://github.com/bbc/psammead/pull/769) Fix stories not appearing in storybook when using `install:packages:link` |
 | 1.1.3 | [PR#775](https://github.com/bbc/psammead/pull/775) Add section to readme regarding using this component without a wrapper and padding/margin |
 | 1.1.2 | [PR#687](https://github.com/bbc/psammead/pull/687) Change visited text colour to `C_CLOUD_DARK` for `C_METAL` |
 | 1.1.1 | [PR#761](https://github.com/bbc/psammead/pull/761) Link default `color` should be `Ebon` not `Postbox` |
-| 1.1.0 | [PR#719](https://github.com/bbc/psammead/pull/719) Updates the link styling to new UX designs |
+| 1.1.0 | [PR#707](https://github.com/bbc/psammead/pull/707) Updates the link styling to new UX designs |
```
Comment (pjlee11) [inline packages/components/psammead-inline-link/CHANGELOG.md:17]: Incorrect link from previous

**Comment by pjlee11** [comment]
New styling in hover/focus state:

image

**Review by jamesdonoh — APPROVED** [review]
👍

**Review by 12 — APPROVED** [review]
LGTM 👍

**Review by dr3 — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**